crd-generator/src/main/java/io/strimzi/crdgenerator/DocGenerator.java

```diff
@@ -272,8 +276,27 @@ private void appendPropertyType(Crd crd, PropertyType propertyType, String exter
     }
 
     private void appendDescription(Class<?> cls) throws IOException {
+        DescriptionFile descriptionFile = cls.getAnnotation(DescriptionFile.class);
         Description description = cls.getAnnotation(Description.class);
-        if (description != null) {
+
+        if (descriptionFile != null)    {
+            String filename = "descriptions/" + cls.getCanonicalName() + ".adoc";
+            InputStream is = cls.getClassLoader().getResourceAsStream(filename);
+
+            if (is == null) {
+                throw new RuntimeException("Class " + cls.getCanonicalName() + " has @DescribeFile annotation, but file " + filename + " does not exist!");
+            }
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(is));
+
+            reader.lines().forEach(line -> {
+                try {
+                    out.append(line.subSequence(0, line.length())).append(NL);
+                } catch (IOException e) {
+                    e.printStackTrace();
+                }
+            });
+        } else if (description != null) {
```

Comment:
If the adoc is in classpath resources then we end up bloating the API jar with documentation which is not needed at runtime. It should be possible to read the documentation from a file within the project which does not get added to the jar (say in `src/main/adoc`).